crates/subspace-networking/src/connected_peers.rs

```diff
@@ -79,13 +79,26 @@ enum ConnectionState {
     NotInterested,
 }
 
+impl ToString for ConnectionState {
```

Comment:
I do not think `ToString` is an appropriate thing here, it looses information, not converting data structure into string representation. Consider something like `impl ConnectionState { fn state_str(&str) -> &'static str {} }` instead.

When I see `ToString` I expect it to be technically possible to have symmetrical `FromStr`, but here it is not the case.